R/neatsort.R

```diff
@@ -0,0 +1,186 @@
+        if (is.numeric(subset) && any(subset > nrow(x))) {
+            stop("Subset refers to rows that do not exist in the data.")
+        } else if (is.character(subset) && any(!subset %in% rownames(x))) {
+            stop("Subset refers to row names that do not exist in the data.")
+        } else if (!is.numeric(subset) && !is.character(subset)) {
+            stop("Subset must be a vector of row indices or names.")
+        }
+    }
+    
+    # Check dimensions are valid
+    if (any(dimensions > ncol(x))) {
+        stop("dimensions refer to columns that do not exist in the data.")
+    }
+    
+    # Check dimension vector is of length 2
+    if (length(dimensions) != 2) {
+        stop("Exactly two dimensions must be specified.")
+    }
+    
+    # Check centering_method
+    centering_method <- match.arg(centering_method, c("mean", "median", "none"))
+    
+    # Check sorting_order
+    sorting_order <- match.arg(sorting_order, c("ascending", "descending"))
+    
+    # Check for unique row names
+    if (any(duplicated(rownames(x)))) {
+        stop("Row names of the matrix must be unique.")
+    }
+    
+    # Check for unique column names
+    if (any(duplicated(colnames(x)))) {
+        stop("Column names of the matrix must be unique.")
+    }
+}
+
+
+#' @description Sorts a matrix by radial theta angle.
+#' @param x A matrix containing the ordinated data to be sorted.
+#' @param subset A vector specifying a subset of rows to be used and retained.
+#' @param dimensions A vector of length 2 specifying the columns of the matrix to use for the X and Y coordinates.
+#' @param centering_method A character string specifying the method to center the data.
+#' @param sorting_order A character string specifying the order of sorting.
+#' @return A vector of row names in the sorted order.
+#' @rdname neatsort
+#' @export
+setMethod("neatsort", signature = c("matrix"),
+          function(x,
+                   subset = NULL,
+                   dimensions = c(1, 2),
+                   centering_method = c("mean", "median", "none"),
+                   sorting_order = c("ascending", "descending"),
+                   ...){
+              
+                # Check args
+                .check_neatsort_args(x, subset, dimensions, centering_method, sorting_order)
+              
+                # Create subset if required
+                if( !is.null(subset) ){
+                    x <- .take_subset(x, subset)
+                }
+
+                # Take the correct dimensions
+                x <- .take_dimensions(x, dimensions)
+
+                # Get the theta values and order them
+                theta_values <- .radial_theta(x, centering_method)
+                ordering <- .get_sorted_rownames(theta_values, sorting_order)
+
+                return(ordering)
+          }
+    )
+
+
+# Takes a subset of rows from the data matrix.
+.take_subset <- function(data, subset) {
+    data <- data[subset, , drop = FALSE]
+    return(data)
+}
```

Comment:
Avoid this kind of functions. As this is just one line, this function is overkill